modern/src/reports/ReportFilter.js

```diff
@@ -0,0 +1,116 @@
+import React, { useState } from 'react';
+import { FormControl, InputLabel, Select, MenuItem, Button, TextField } from '@material-ui/core';
+import t from '../common/localization';
+import { useSelector } from 'react-redux';
+import moment from 'moment';
+
+const ReportFilter = (props) => {
```

Comment:
Can we update route report to use this? Also, how can we add additional filed here? For example for events report we need to filter by event type.

Reply:
I'll update route report using this.

I think we can use reportType attribute for showing custom fields in the ReportFilter component. Like Below-

<ReportFilter reportType = 'event|route|...' handleSubmit={handleSubmit} />

{ reportType && reportType === 'event' && (
	<FormControl variant="filled" margin="normal" fullWidth>
		<InputLabel>Event Type</InputLabel>
		<Select value={eventType} onChange={(e) => setEventType(e.target.value)}>
			<MenuItem value="ignitionOn">{t('eventIgnitionOn')}</MenuItem>
			<MenuItem value="ignitionOff">{t('eventIgnitionOff')}</MenuItem>
		</Select>
	 </FormControl>
  )
}

Reply:
I think the right solution is to provide a way for each report to add custom fields in addition to base ones.

Reply:
I've added `FilterForm` that you can use here.